api/authorization-api-1_0.md

```diff
@@ -888,165 +889,97 @@ The following is a non-normative example of a response to an Access Evaluations
-    }
-  ],
-  "page": {
-    "next_token": ""
-  }
-}
-~~~
+`properties`:
+: OPTIONAL. An object containing additional pagination request attributes, such as, but not limited to, sorting and filtering.
 
-### Paged requests
+Additional keys MAY be included in the `page` object. If they are, they MUST be defined in a specification referenced in the AuthZEN Policy Decision Point Capabilities Registry ({{iana-pdp-capabilities-registry}}). Furthermore, the PDP MUST declare support for the corresponding capability URN in its `supported_capabilities` metadata ({{pdp-metadata-data-capabilities}}).
 
-A response that needs to be split across page boundaries returns a non-empty `page.next_token`.
+### Paginated Responses {#search-pagination-response}
 
-#### Example
+Any Search Response MAY include a `page` object, but if a response does not contain the entire result set, it MUST include this object.
 
-~~~ json
-{
-  "results": [
-    {
-      "type": "user",
-      "id": "alice@example.com"
-    },
-    {
-      "type": "user",
-      "id": "bob@example.com"
-    }
-  ],
-  "page": {
-    "next_token": "alsehrq3495u8"
-  }
-}
-~~~
+The `page` object contains the following keys:
 
-To retrieve the next page, provide `page.next_token` in the next request:
+`next_token`: 
+: REQUIRED. An opaque string value indicating the next page of results to return. If there are no more results after this page, its value MUST be an empty string.
 
-~~~ json
-{
-  "subject": {
-    "type": "user"
-  },
-  "action": {
-    "name": "can_read"
-  },
-  "resource": {
-    "type": "account",
-    "id": "123"
-  },
-  "context": {
-    "time": "2024-10-26T01:22-07:00"
-  },
-  "page": {
-    "next_token": "alsehrq3495u8"
-  }
-}
-~~~
+`count`: 
+: OPTIONAL. The number of results included in this response. When included at the start of a response, as described in the Search API Response ({{search-response}}), this enables clients to display progress indicators when processing large or slow responses.
 
-Note: page size is implementation-dependent.
+`total`:
+: OPTIONAL. The total number of results in the result set.
 
-# Resource Search API {#resource-search-api}
+`properties`:
+: OPTIONAL. An object containing additional pagination response attributes. Examples include, but are not limited to, estimated totals or the number of remaining results.
 
-The Resource Search API defines the message exchange pattern between a client (PEP) and an authorization service (PDP) for returning all of the resources that match the search criteria.
+### Offset capability {#search-pagination-offset}
 
-The Resource Search API is based on the Access Evaluation information model, but omits the Resource ID.
+Using the offset capability, clients can indicate the number of results to skip before starting to return results. This enables clients to interactively navigate result sets and, when combined with the Limit capability ({{search-pagination-limit}}), can support classic page-based navigation.
```

Comment:
My general comment is that the spec should cleanly separate the base mechanism (paging using opaque tokens, which is guaranteed to work across implementations), and the "advanced paging" mechanism.  The former should have non-normative examples. The latter should have non-normative examples in a separate section.  

This makes it much easier for spec readers to implement a client of their choosing.